source/Xunit.Gherkin.Quick/CoreModel/FeatureFile/FeatureFile.cs

```diff
@@ -13,11 +13,16 @@ public FeatureFile(GherkinDocument gherkinDocument)
             GherkinDocument = gherkinDocument ?? throw new System.ArgumentNullException(nameof(gherkinDocument));
         }
 
-        public global::Gherkin.Ast.Scenario GetScenario(string scenarioName)
+        public global::Gherkin.Ast.ScenarioDefinition GetScenario(string scenarioName)
         {
-            return GherkinDocument.Feature.Children.FirstOrDefault(s => s.Name == scenarioName) as global::Gherkin.Ast.Scenario;
+            return GherkinDocument.Feature.Children.FirstOrDefault(s => s.Name == scenarioName);
         }
 
+		public global::Gherkin.Ast.ScenarioDefinition GetBackgroundScenario()
+		{
+			return GherkinDocument.Feature.Children.SingleOrDefault(s => s.Keyword == "Background");
```

Comment:
It will be safer if you depend on the concrete type of the ScenarioDefinition. i.e. return those which are of type `Background`. That would be much safer comparison without involving string keyword.